### PR TITLE
Add missing table features when adding `timestamp_ntz` column in Delta Lake

### LINKED ISSUE #24158: Adding timestamp_ntz column should add reader and writer features
Trino:
```sql
CREATE TABLE test (a int) WITH (change_data_feed_enabled = true);
TABLE "test$properties";
             key             | value
-----------------------------+-------
 delta.enableChangeDataFeed  | true
 delta.enableDeletionVectors | false
 delta.minReaderVersion      | 1
 delta.minWriterVersion      | 4
(4 rows)

-- This ADD COLUMN should add delta.feature.changeDataFeed = supported
ALTER TABLE test ADD COLUMN b timestamp;
TABLE "test$properties";
             key             |   value
-----------------------------+-----------
 delta.enableDeletionVectors | false
 delta.enableChangeDataFeed  | true
 delta.minReaderVersion      | 3
 delta.minWriterVersion      | 7
 delta.feature.timestampNtz  | supported
```

Spark:
```sql
CREATE TABLE test (a int) USING delta TBLPROPERTIES (delta.enableChangeDataFeed = true) LOCATION 's3://test-bucket/test';
SHOW TBLPROPERTIES test;
delta.enableChangeDataFeed	true
delta.minReaderVersion	1
delta.minWriterVersion	4

-- SET TBLPROPERTIES is required before adding timestamp_ntz column
ALTER TABLE test SET TBLPROPERTIES ('delta.feature.timestampNtz' = 'supported');
ALTER TABLE test ADD COLUMN b timestamp_ntz;
delta.enableChangeDataFeed	true
delta.feature.appendOnly	supported
delta.feature.changeDataFeed	supported
delta.feature.checkConstraints	supported
delta.feature.generatedColumns	supported
delta.feature.invariants	supported
delta.feature.timestampNtz	supported
delta.minReaderVersion	3
delta.minWriterVersion	7
```

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Fixes #24158

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

**Comment by ebyhr** [comment]
`TestDeltaLakeAlterTableCompatibility#testTrinoPreservesTableFeature` started failing.

**Comment by ebyhr** [comment]
/test-with-secrets sha=b200a57d9d3083f64043026e2f69f7c7e54e16f2

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java [hunk 1]
```diff
@@ -389,16 +384,14 @@ public class DeltaLakeMetadata
...
-    private static final int MIN_WRITER_FEATURE_VERSION = 7;
-    private static final int CDF_SUPPORTED_WRITER_VERSION = 4;
-    private static final int COLUMN_MAPPING_MODE_SUPPORTED_READER_VERSION = 2;
-    private static final int COLUMN_MAPPING_MODE_SUPPORTED_WRITER_VERSION = 5;
-    private static final int TIMESTAMP_NTZ_SUPPORTED_READER_VERSION = 3;
-    private static final int TIMESTAMP_NTZ_SUPPORTED_WRITER_VERSION = 7;
-    private static final int DELETION_VECTORS_SUPPORTED_READER_VERSION = 3;
-    private static final int DELETION_VECTORS_SUPPORTED_WRITER_VERSION = 7;
+
+    public static final int CDF_SUPPORTED_WRITER_VERSION = 4;
```
Comment (pajaks) [inline plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java:388]: Maybe those should be moved to `DeltaLakeTableFeatures` to be in the same place as feature name and min version

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/ProtocolEntry.java [hunk 2]
```diff
@@ -61,4 +73,73 @@ public boolean writerFeaturesContains(String featureName)
...
+        private int readerVersion;
+        private int writerVersion;
+        private final Set<String> readerFeatures = new HashSet<>();
+        private final Set<String> writerFeatures = new HashSet<>();
+
+        public Builder(int readerVersion, int writerVersion, Optional<Set<String>> readerFeatures, Optional<Set<String>> writerFeatures)
+        {
+            this.readerVersion = readerVersion;
+            this.writerVersion = writerVersion;
+            requireNonNull(readerFeatures, "readerFeatures is null").ifPresent(this.readerFeatures::addAll);
```
Comment (pajaks) [inline plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/ProtocolEntry.java:98]: It adds feature without checking if `readerVersion` and `writerVersion` as in separate `enable....` Should caller of Builder assure that those version are consistent with features?
Or maybe this constructor should be made private?

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java [hunk 3]
```diff
@@ -4201,6 +4202,38 @@ public void testQueriesWithoutCheckpointFiltering()
...
+                    .containsExactlyInAnyOrderEntriesOf(ImmutableMap.<String, String>builder()
+                            .put("delta.enableChangeDataFeed", "true")
+                            .put("delta.enableDeletionVectors", "false")
+                            .put("delta.minReaderVersion", "1")
+                            .put("delta.minWriterVersion", "4")
+                            .buildOrThrow());
+
+            assertUpdate("ALTER TABLE " + table.getName() + " ADD COLUMN ts TIMESTAMP");
+
+            assertThat(getTableProperties(table.getName()))
```
Comment (pajaks) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java:4219]: Please add comment why `delta.feature.changeDataFeed` is now supported

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java (hunk 1 above)
Comment (ebyhr) [reply plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java:388]: I would ignore this suggestion. This constant is mainly for `delta.enableChangeDataFeed` configuration, not table features.

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java [hunk 4]
```diff
@@ -4201,6 +4202,39 @@ public void testQueriesWithoutCheckpointFiltering()
...
+                    .containsExactlyInAnyOrderEntriesOf(ImmutableMap.<String, String>builder()
+                            .put("delta.enableChangeDataFeed", "true")
+                            .put("delta.enableDeletionVectors", "false")
+                            .put("delta.minReaderVersion", "1")
+                            .put("delta.minWriterVersion", "4")
+                            .buildOrThrow());
+
+            assertUpdate("ALTER TABLE " + table.getName() + " ADD COLUMN ts TIMESTAMP");
+
+            // CDF is enabled in this table. 'delta.feature.changeDataFeed' must  be added when updating the table to versions supporting table features
```
Comment (raunaqmorarka) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java:4219]: nit: extra space in comment after `must`

**Review by raunaqmorarka — APPROVED** [review]

**Pull request merged**